composer.json

```diff
@@ -55,7 +55,7 @@
         "php-cs-fixer/accessible-object": "^1.1",
         "php-cs-fixer/phpunit-constraint-isidenticalstring": "^1.6",
         "php-cs-fixer/phpunit-constraint-xmlmatchesxsd": "^1.6",
-        "phpunit/phpunit": "^9.6.22 || ^10.5.45 || ^11.5.7",
+        "phpunit/phpunit": "^9.6.22 || ^10.5.45 || ^11.5.7 || ^12.0.2",
```

Comment:
tbh, i wouldn't claim PHPUnit 12 compatibility if it will fail without running `dev-tools/.php-cs-fixer.php-unit-attributes.php` migration.

Reply:
With, in practice, only converting PHPDocs to attributes, I'd say it is compatible - we are sure that no other BC changes are making the tests fail.

Maybe it is time to think about dropping the support for PHP 7.4? It will make many of the gymnastics we make to support PHP 7.4 to 8.4 go away. It 2 years since it lost security support.

Reply:
John Doe opens the project, runs `composer install`. PHPUnit 12 is installed.
they run `./vendor/bin/phpunit` and tests are crashing all over the place.
This is far from msg that PHPUnit 12 is supported as test runner in the project.

Reply:
> John Doe opens the project, runs `composer install`. PHPUnit 12 is installed.

That's not happening right now (I mean `git clone` + `composer update` on the main branch), but might happen any time.